### ISSUE TITLE
Tests fail if Zonemaster::LDNS was compiled without IDN support

### ISSUE THREAD

**Issue by mattias-p** [dotse/zonemaster-backend#235]
If IDN support is missing Zonemaster Backend should either skip tests that depend on IDN or the requirement for IDN should be clearly stated in the documentation.

```
$ make test
PERL_DL_NONLAZY=1 /usr/local/bin/perl "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'inc', 'blib/lib', 'blib/arch')" t/*.t
t/test01.t ................ ok
t/test_validate_syntax.t .. 1/?
#   Failed test 'idn domain=[é]'
#   at t/test_validate_syntax.t line 34.
# Domain name contains non-ascii characters and IDN conversion is not installed

#   Failed test 'idn domain=[éé]'
#   at t/test_validate_syntax.t line 39.
# Domain name contains non-ascii characters and IDN conversion is not installed

#   Failed test 'idn domain=[é]'
#   at t/test_validate_syntax.t line 88.
# NS [&eacute;] contains non-ascii characters and IDN conversion is not installed

#   Failed test 'idn domain=[éé]'
#   at t/test_validate_syntax.t line 93.
# NS [&eacute;&eacute;] contains non-ascii characters and IDN conversion is not installed
# Looks like you failed 4 tests of 27.
t/test_validate_syntax.t .. Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/27 subtests

Test Summary Report
-------------------
t/test_validate_syntax.t (Wstat: 1024 Tests: 27 Failed: 4)
  Failed tests:  4-5, 12-13
  Non-zero exit status: 4
Files=2, Tests=54, 11 wallclock secs ( 0.04 usr  0.00 sys + 10.88 cusr  0.30 csys = 11.22 CPU)
Result: FAIL
Failed 1/2 test programs. 4/54 subtests failed.
*** Error code 4

Stop.
make: stopped in /usr/home/mattiasp/zonemaster-backend
```

**Comment by mtoma** [comment]
I think this requirement should be added to the documentation.

----- Original Message -----

**Comment by matsduf** [comment]
Since it clearly is important for the features of the backend, I suggest that requiring IDN support is default and then the current tests should pass. There should be a possibility to install the backend without IDN support, and then those tests should be skipped.

And yes, this should be documented too.

**Comment by sandoche2k** [comment]
@mattias-p https://github.com/dotse/zonemaster-ldns/issues/24

**Comment by sandoche2k** [comment]
Closed by dotse/zonemaster-ldns#24
